src/components/Messaging/Survey.js

```diff
@@ -0,0 +1,385 @@
+  Select,
+  FormControl,
+  OutlinedInput,
+} from '@material-ui/core';
+import { Close } from '@material-ui/icons';
+import { makeStyles } from '@material-ui/styles';
+
+import GSInputLabel from 'components/common/InputLabel';
+import { MessagingContext } from 'context/MessagingContext';
+import { AppContext } from 'context/AppContext';
+
+const DRAWER_WIDTH = 300;
+
+const useStyles = makeStyles((theme) => ({
+  drawer: {
+    background: 'light tan',
+    width: DRAWER_WIDTH,
+    padding: theme.spacing(3),
+  },
+  title: {
+    width: '100%',
+    display: 'flex',
+    flexFlow: 'row',
+  },
+  formTitle: {
+    color: theme.palette.primary.main,
+  },
+  directions: {
+    color: 'grey',
+    margin: '5px',
+    wordWrap: 'break-word',
+  },
+  form: {
+    width: DRAWER_WIDTH - 20,
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    marginBottom: '2em',
+  },
+  input: {
+    margin: '5px',
+    '& .MuiInputBase-input': {
+      fontSize: 16,
+      padding: '10px 12px',
+    },
+  },
+  submitButton: {
+    margin: '10px auto',
+    width: DRAWER_WIDTH - 50,
+    background: theme.palette.primary.main,
+    color: 'white',
+  },
+  surveyCloseButton: {
+    marginLeft: 'auto',
+    border: '1px solid ' + `${theme.palette.primary.main}`,
+  },
+}));
+
+const SurveyForm = () => {
+  const { form, submitButton, input } = useStyles();
+  const { user, regions, postMessageSend } = useContext(MessagingContext);
+  const { orgList } = useContext(AppContext);
+  const [survey, setSurvey] = useState({
+    questionOne: '',
+    choiceOne: '',
+    choiceTwo: '',
+    choiceThree: '',
+    questionTwo: '',
+    choiceFour: '',
+    choiceFive: '',
+    choiceSix: '',
+    questionThree: '',
+    choiceSeven: '',
+    choiceEight: '',
+    choiceNine: '',
+    organization: '',
+    region: '',
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setSurvey({
+      ...survey,
+      [name]: value,
+    });
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    let one = { prompt: '', choices: [] };
+    let two = { prompt: '', choices: [] };
+    let three = { prompt: '', choices: [] };
+
+    // Map through survey info to match api spec
+    Object.entries(survey).map(([key, val]) => {
+      switch (key) {
+        case 'questionOne':
+          one.prompt = val;
+          break;
+        case 'choiceOne':
+        case 'choiceTwo':
+        case 'choiceThree':
+          one.choices.push(val);
+          break;
+        case 'questionTwo':
+          two.prompt = val;
+          break;
+        case 'choiceFour':
+        case 'choiceFive':
+        case 'choiceSix':
+          two.choices.push(val);
+          break;
+        case 'questionThree':
+          three.prompt = val;
+          break;
+        case 'choiceSeven':
+        case 'choiceEight':
+        case 'choiceNine':
+          three.choices.push(val);
+          break;
+      }
```

Comment:
Should this switch statement have a default clause?

Reply:
I am going to get rid of the switch statement all together just coming up with a better solution